src/Chapter2.hs

```diff
@@ -842,7 +880,11 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [Int] -> [Int]
+rotate n l
+  | length l == 0 = []
+  | n < 0 = []
+  | otherwise = drop n (take (length l + n) (cycle l))
```

Comment:
That's a correct solution! Though, it does a little bit more than work when the given number is big and the list is small. You will cycle through the list a lot. Maybe you can think of a way to optimize this :slightly_smiling_face:

Reply:
Cool! Thank you for suggestion. I found that rotating array with 6 element by 34 elements is the same that rotating by 4 elements. I have rewritten this function